src/autorest-core/lib/data-store/data-store.ts

```diff
@@ -320,8 +321,28 @@ export class DataStore extends DataStoreView {
     return Object.getOwnPropertyNames(this.store);
   }
 
-  public Blame(absoluteUri: string, position: SmartPosition): BlameTree {
+  public Blame(absoluteUri: string, position: SmartPosition, message?: (m: Message) => void): BlameTree {
     const data = this.ReadStrictSync(absoluteUri);
+
+    // verify if the path exists in the document
+    if ('path' in position) {
+      let x = (<any>JSON.parse(data.ReadData()));
```

Comment:
raw data can also be YAML and what not - `ReadObject` will always do the right thing so use that instead of `ReadData`